doc/modules/impute.rst

```diff
@@ -120,3 +120,46 @@ whether or not they contain missing values::
          [False,  True, False, False]])
   >>> indicator.features_
   array([0, 1, 2, 3])
+
+When using it in a pipeline, be sure to use the :class:`FeatureUnion` to add
+the indicator features to the regular features. First we obtain the `audiology`
+dataset from OpenML, which has many missing values.
+
+  >>> from sklearn.datasets import fetch_openml
+  >>> from sklearn.impute import SimpleImputer, MissingIndicator
+  >>> from sklearn.model_selection import train_test_split
+  >>> from sklearn.pipeline import FeatureUnion, make_pipeline
+  >>> from sklearn.tree import DecisionTreeClassifier
+  >>> X, y = fetch_openml('audiology', 1, return_X_y=True)
```

Comment:
I would prefer an example that doesn't require an internet connection + download

Reply:
I guess we don't have any build-in datasets with missing values? Should we add one? Should we have build-in titanic or something? Having one built-in dataset with mixed types and missing values might be nice.

Reply:
I'd be more comfortable just inserting missing values to illustrate the point. We usually don't worry about real world in user guide